src/main/java/pl/piekoszek/gorskimatches/challange/ChallengeController.java

```diff
@@ -0,0 +1,99 @@
+package pl.piekoszek.gorskimatches.challange;
+
+import org.springframework.web.bind.annotation.*;
+import pl.piekoszek.gorskimatches.equation.EquationRandomizer;
+import pl.piekoszek.gorskimatches.token.EmailService;
+
+import java.util.List;
+import java.util.UUID;
+import java.util.stream.Collectors;
+
+@RestController
+@RequestMapping("api/challenge")
+public class ChallengeController {
+
+    private final EquationRandomizer equationRandomizer;
+
+    private final GenerateUUID generateUUID;
+
+    private EmailService emailService;
+
+    private ChallengeRepository challengeRepository;
+
+    public ChallengeController(EquationRandomizer equationRandomizer, GenerateUUID generateUUID, ChallengeRepository challengeRepository, EmailService emailService) {
+        this.equationRandomizer = equationRandomizer;
+        this.generateUUID = generateUUID;
+        this.challengeRepository = challengeRepository;
+        this.emailService = emailService;
+    }
+
+    @GetMapping("generate")
+    UUID getUUID() {
+        var challenge = new Challenge();
+        challenge.setChallengeQuizzes(
+                equationRandomizer.equationsForChallenge().stream()
+                        .map(quiz -> new ChallengeQuiz(challenge, quiz))
+                        .collect(Collectors.toList()));
+        challenge.setUuid(generateUUID.generateUUID());
+        challengeRepository.save(challenge);
+        return challenge.getUuid();
+    }
+
+    @PostMapping("score")
+    void getScore(@RequestBody Challenge challenge) {
```

Comment:
I think this method request body should be like:
```
{
   "uuid": "12319u42394...",
   "score": "10,
   "time": 20
}
```
And to see if it's registered user, or not, see JWT Token.

I'd suggest extending `@interface Email` to accept `required` parameter. So that it would throw unauthorized exception only if `required=true`.